src/repository/screens/commit.screen.js

```diff
@@ -0,0 +1,280 @@
+import { translate } from 'utils';
+import { colors, fonts, normalize } from 'config';
+import { getCommitDetails } from '../repository.action';
+
+const mapStateToProps = state => ({
+  language: state.auth.language,
+  commit: state.repository.commit,
+  diff: state.repository.diff,
+  isPendingCommit: state.repository.isPendingCommit,
+  isPendingDiff: state.repository.isPendingDiff,
+});
+
+const mapDispatchToProps = dispatch => ({
+  getCommitDetailsByDispatch: commit => dispatch(getCommitDetails(commit)),
+});
+
+const styles = StyleSheet.create({
+  fileChangeContainer: {
+    padding: 0,
+    marginTop: 12,
+    marginBottom: 12,
+  },
+  fileTitleContainer: {
+    flexDirection: 'row',
+    paddingVertical: 15,
+    backgroundColor: colors.greyVeryLight,
+  },
+  linesChanged: {
+    flex: 0.3,
+    paddingLeft: 10,
+    flexDirection: 'row',
+    alignItems: 'center',
+  },
+  lineNumbersChanged: {
+    ...fonts.fontCode,
+    marginRight: 5,
+  },
+  fileTitle: {
+    flex: 1,
+    marginLeft: 10,
+  },
+  codeStyle: {
+    ...fonts.fontCode,
+    fontSize: normalize(10),
+  },
+  dividerStyle: {
+    marginBottom: 0,
+  },
+  noChangesMessage: {
+    ...fonts.fontPrimarySemiBold,
+    paddingVertical: 5,
+    paddingLeft: 10,
+  },
+  newIndicator: {
+    ...fonts.fontPrimarySemiBold,
+    color: colors.green,
+  },
+  deletedIndicator: {
+    ...fonts.fontPrimarySemiBold,
+    color: colors.red,
+  },
+  headerContainer: {
+    paddingTop: 15,
+    paddingHorizontal: 25,
+  },
+  header: {
+    flex: 1,
+    paddingTop: 10,
+    flexDirection: 'row',
+    alignItems: 'center',
+    justifyContent: 'space-between',
+  },
+  headerItem: {
+    flex: 1,
+  },
+  headerText: {
+    ...fonts.fontPrimarySemiBold,
+    fontSize: normalize(14),
+  },
+});
+
+class Commit extends Component {
+  props: {
+    navigation: Object,
+    commit: Object,
+    diff: string,
+    isPendingCommit: boolean,
+    isPendingDiff: boolean,
+    getCommitDetailsByDispatch: Function,
+    language: string,
+  };
+
+  componentDidMount() {
+    const { navigation, getCommitDetailsByDispatch } = this.props;
+    const commit = navigation.state.params.commit;
+
+    getCommitDetailsByDispatch(commit);
+  }
+
+  keyExtractor = (item, index) => {
+    return index;
+  };
+
+  renderHeader = () => {
+    const { commit, language, isPendingCommit } = this.props;
+    const message = commit.commit ? commit.commit.message : 'Loading...';
+    const committer = commit.author ? commit.author.login : '';
+
+    if (isPendingCommit || !commit.files) {
+      return (
+        <Text>
+          {message}
+        </Text>
+      );
+    }
+
+    return (
+      <View style={styles.headerContainer}>
+        <View style={styles.header}>
+          <Text>
+            {message}
+          </Text>
+        </View>
+        <View style={styles.header}>
+          <Text>
+            {translate('repository.commit.byConnector', language, {
+              contributor: committer,
```

Comment:
Can we make a clickable committer? To open a profile.

Reply:
Good idea! I add it to the checklist 😃👍